core/src/androidMain/kotlin/Filter.kt

```diff
@@ -20,6 +20,16 @@ public actual sealed class Filter {
         public val dataMask: ByteArray?,
     ) : Filter()
 
+    /**
+     * https://developer.android.com/reference/android/bluetooth/le/ScanFilter.Builder#setDeviceAddress(java.lang.String)
+     * https://developer.android.com/reference/android/bluetooth/le/ScanFilter.Builder#setDeviceName(java.lang.String)
+     */
+    public class DeviceInfo(
+        public val address: String? = null,
+
+        public val name: String? = null,
+    ) : Filter()
```

Comment:
I think it makes sense to have these as individual classes so that their values don't need to be `null`able.

```suggestion
    /**
     * https://developer.android.com/reference/android/bluetooth/le/ScanFilter.Builder#setDeviceName(java.lang.String)
     */
    public actual data class Name actual constructor(
        public val name: String,
    ) : Filter()

    /**
     * https://developer.android.com/reference/android/bluetooth/le/ScanFilter.Builder#setDeviceAddress(java.lang.String)
     */
    public data class Address(
        public val address: String,
    ) : Filter()
```